Modification to adapt to the installation of the Cdp of Strasbourg

ClemClem25 opened the pull request:
Hi,

I've created this Pull Request because in Strasbourg, we have a different installation than the one in Marseille. Since February, I've tested these modifications and haven't encountered any major issues.

**Modifications overview :** 

**omm_conditioner:**
We have a pump for rewards, which is activated with serial.write(pm) and stopped when we send p0. The goal was to add another option called Juice_pump, where you can choose the duration and send any character you want to start or stop the pump, so it can work with different criteria.

**omm_detect_participant:**
We use two antennas to detect the RFID, as participants have one on each arm. The code now allows you to select two COM ports.

I've never created a Pull Request before, so if you need any additional information, feel free to ask any questions. (Also, I used ChatGPT and Mistral AI to modify parts of the code.)

smathot commented:
Thanks for this! I'll get back to you next week.

smathot commented:
It's great that you're working on this! Maintaining this code for an additional site goes beyond my responsibilities, so let's make sure that we're all on the same page with how to move forward with this.

@nclaidiere Can you confirm that you would like @ClemClem25 to work on this? It will be your responsibility to test his updates in Rousset? In principle, it seems like these are low-risk modifications, but they do need to be tested, mainly to make sure that new functionality implemented for Strasbourg doesn't break existing functionality for Rousset.

@ClemClem25 To reduce the chances of introducing bugs, a few matters of style:

- Separate this pull request so that there is one for `omm_conditioner` and one for `omm_detect_participant`. These should non-overlapping PRs referenced to the current master branch.
- Do not use French code comments or names.
- Do not change large chunks of code without making clear why you did so. In this case you entirely reimplemented `_rfid_monitor()`, but it's not clear to me why. It's ok to use AI to write code, but make sure you understand what the code does, and only change what needs to be changed.

If this works for both of you, then I will guide you through the process, make sure that no obvious things go wrong, but I will not extensively test things.

ClemClem25 commented:
Hi Sebastiaan,

Thank you for your feedback and guidance on this process.

If Nicolas Claidière is okay with handling the testing, I’ll go ahead and revise the code to ensure it only includes the necessary changes, with comments and variable names in English. I’ll also separate the pull request, one for omm_conditioner and one for omm_detect_participant.

smathot commented:
> I am a bit worried about removing updates but we will see, I'll ask for help!

What do you mean with removing updates?

ClemClem25 commented:
Hi @nclaidiere, @smathot, 
I updated the code in omm_conditioner and omm_detect_participant to remove French code and comments. I also simplified the modifications in omm_detect_participant to better preserve the original structure.

Now I need to split this into two separate PRs. Should I close this one and open two new ones one for omm_conditioner and one for omm_detect_participant to proceed correctly?

smathot commented:
> Now I need to split this into two separate PRs. Should I close this one and open two new ones one for omm_conditioner and one for omm_detect_participant to proceed correctly?

Yes, so that will require a bit of manual copy-pasting.

smathot commented:
@ClemClem25 Actually, for now, let's not split the PR. Doing so would be mainly for good style, but it also takes time and isn't really necessary. Let's move on to testing. 

1. Change the version number in pyproject.toml. Use an alpha version, such as 0.7.1a1, so that later we can increment to 0.7.1. 
2. Build a wheel (Python package) using `poetry build`, which is a command line utility that you need to install.
3. Install the wheel on your own test system, and give it to @nclaidiere to do the same on one of his.
4. Make sure that you're actually running the updated wheel.
5. Carefully check that nothing has broken.
6. Report back.

ClemClem25 commented:
Thank you, @smathot , for this detailed plan of action.

I created the link and have sent it to @nclaidiere. Before doing so, I installed the updated wheel on the modules to test it, and for now, everything is running well.

I will keep you updated!

Will keep you updated !

gdesor commented:
Hello, I have also modified the omm-client plugin to integrate one new RFID cards and a new control board for reward delivery.
I think it would be a good idea to structure the code so that any future changes are easier for everyone and code review can be faster.

I’d like to propose restructuring omm_detect_participant similarly to how omm_conditionner is organized — with a dedicated subdirectory for each detection method (e.g., Form, Keypress, RFID Card 1, RFID Card N, etc.). This structure would make it easier for everyone to contribute or maintain their own modules without interfering with others’, and would simplify future development and code reviews. What do you think?

You can check the modification at https://github.com/gdesor/omm-client-sdp/tree/multi-devices (multidevices branch)

My main modifications are described in [the changelog.md file.](https://github.com/gdesor/omm-client-sdp/blob/multi-devices/changelog.md?plain=1)

@ClemClem25 I’ve integrated your code, I’m transferring the wheel on Mattermost.

smathot commented:
@gdesor I agree with this restructuring. Can you open a separate PR against the master branch? Before you do, remove the changelog file. You can describe the changes in the PR itself, and of course in the commits.
